assets/js/data/cart/actions.js

```diff
@@ -181,3 +194,33 @@ export function* removeItemFromCart( cartItemKey ) {
 
 	yield itemQuantityPending( cartItemKey, false );
 }
+
+/**
+ * Changes the quantity for specified cart item:
+ * - Calls API to set quantity.
+ * - If successful, yields action to update store.
+ * - If error, yields action to store error.
+ * - Sets cart item as pending while API request is in progress.
+ *
+ * @param {string} cartItemKey Cart item being updated.
+ * @param {number} quantity Specified (new) quantity.
+ */
+export function* changeCartItemQuantity( cartItemKey, quantity ) {
+	yield itemQuantityPending( cartItemKey, true );
+
+	try {
+		const result = yield apiFetch( {
+			path: `/wc/store/cart/items/${ cartItemKey }?quantity=${ quantity }`,
+			method: 'PUT',
+			cache: 'no-store',
+		} );
```

Comment:
This might change pending the decision in #1840 with how cart gets updated. But for now we can leave.